zeroconf/aio.py

```diff
@@ -0,0 +1,131 @@
+""" Multicast DNS Service Discovery for Python, v0.14-wmcbrine
+    Copyright 2003 Paul Scott-Murphy, 2014 William McBrine
+
+    This module provides a framework for the use of DNS Service Discovery
+    using IP multicast.
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301
+    USA
+"""
+import asyncio
+from typing import Optional
+
+from . import (
+    IPVersion,
+    InterfaceChoice,
+    InterfacesType,
+    REGISTER_TIME,
+    ServiceInfo,
+    UNREGISTER_TIME,
+    Zeroconf,
+)
+
+
+class AsyncZeroconf(Zeroconf):
```

Comment:
Since this is gonna expose an interface different from `Zeroconf` maybe it'll make more sense to use composition rather than inheritance for that? As in `AsyncZeroconf` containing an internal `Zeroconf` instance and celling methods on *that* instance instead of on `self`.

Reply:
I'll make it a composition